src/network/team_switch_request_message.cpp

```diff
@@ -0,0 +1,66 @@
+#include "platform/i_platform.h"
+#include "network/team_switch_request_message.h"
+
+namespace network {
+
+TeamSwitchRequestMessageSenderSystem::TeamSwitchRequestMessageSenderSystem()
+    : MessageSenderSystem()
+{
+    mOnTeamSwitchRequestEvent = platform::EventServer<TeamSwitchRequestEvent>::Get().Subscribe( boost::bind( &TeamSwitchRequestMessageSenderSystem::OnTeamSwitchRequest, this, _1 ) );
+}
+
+
+void TeamSwitchRequestMessageSenderSystem::Init()
+{
+    MessageSenderSystem::Init();
+}
+
+
+void TeamSwitchRequestMessageSenderSystem::Update(double DeltaTime)
+{
+    MessageSenderSystem::Update(DeltaTime);
+}
+
+void TeamSwitchRequestMessageSenderSystem::OnTeamSwitchRequest( TeamSwitchRequestEvent const& event )
+{
+    if ( mProgramState.mMode == ProgramState::Client )
+    {
+        if ( mProgramState.mClientId != event.mClientId )
+        {
+            // ERROR: don't allow change other's team
```

Comment:
well, it's rather bad phrasing. it should be something like "filtering: don't allow changing another client's team"
I'll change that